.github/workflows/tests.yml

```diff
@@ -327,6 +328,33 @@ jobs:
       repository-name: ${{ github.repository }}
       dry-run: false
 
+  build-docker-rootless:
+    uses: ./.github/workflows/build-test-docker.yaml
+    secrets: inherit
+    with:
+      docker-tags: |
+        type=semver,pattern={{version}},suffix=-rootless
```

Comment:
what are all those docker tags for? I don't really understand the syntax but why do we need all
those things?

Reply:
This is the configuration that set different tags on images that get pushed to Dockerhub depending on the context: https://hub.docker.com/r/returntocorp/semgrep/tags. For example, `latest` for every commit to develop, `pr-nnnn` for every pull request, and `sha-nnnnnnnn` for every commit in the repo.

This block uses the same tagging strategy as the root container but adds a suffix `-nonroot` to each tag. In this PR, it results in two images being pushed - one with:
```Pushing image tags...
-> returntocorp/semgrep:pr-8572
-> returntocorp/semgrep:sha-9ed0477
```
and another with:
```
Pushing image tags...
-> returntocorp/semgrep:pr-8572-nonroot
-> returntocorp/semgrep:sha-9ed0477-nonroot
```

Once merged to develop, it'll cause to images to be pushed `returntocorp/semgrep:latest` and `returntocorp/semgrep:latest-nonroot`.